presto-main/src/main/java/com/facebook/presto/execution/LocationFactory.java

```diff
@@ -28,5 +28,10 @@
 
     URI createTaskLocation(InternalNode node, TaskId taskId);
 
+    /**
+     * TODO: this should be merged into createTaskLocation once full thrift support is in-place for v1/task
+     */
+    URI createRemoteTaskLocation(InternalNode node, TaskId taskId);
```

Comment:
I see. So this method is needed because some RPC is supported by thrift while some others still need HTTP? Maybe amend the comment as:
```
    /**
    * TODO: this method is required since not not all RPC call is supported by thrift.
     * .   it should be merged into createTaskLocation once full thrift support is in-place for v1/task
     */
```

Actually, I am a bit confused about when `createTaskLocation` is used and when `createRemoteTaskLocation` is used. Let's quickly talk in person on this.